src/sfctl/tests/app_test.py

```diff
@@ -87,3 +88,81 @@ def parse_app_metrics_single_test(self):
         self.assertEqual(res.maximum_capacity, '3')
         self.assertEqual(res.reservation_capacity, '2')
         self.assertEqual(res.total_application_capacity, '2')
+
+    def parse_cluster_manifest_test(self):
+        """Parse cluster manifest loaded from a file"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'clustermanifest1.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
+                         'fabric:ImageStore')
+
+    def parse_empty_cluster_manifest_test(self): #pylint: disable=invalid-name
+        """Parse empty cluster manifest content"""
+        self.assertEqual(sf_c.parse_imagestore_connection_string(None),
+                         None)
+
+    def parse_missing_imagestore_entry_cluster_manifest_test(self): #pylint: disable=invalid-name
+        """Parse missing image store connection string in cluster manifest"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'invalid_clustermanifest1.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
```

Comment:
You can actually use vcrpy combined with an invalid string to get more thorough test scenarios with less file I/O. Especially if you already have a cluster already. I'd move the cluster manifest parsing test to a scenario test if you want to use a non-trivial string for test parsing.